StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerViewController.swift

```diff
@@ -268,7 +268,28 @@ final class LinkAccountPickerViewController: UIViewController {
                 pane: .linkAccountPicker
             )
 
-        if nextPane == .success {
+        if
+            dataSource.acquireConsentOnPrimaryCtaClick,
+            let selectedAccount = dataSource.selectedAccounts.first(
+                // to better handle multi-select, we want to ensure that
+                // the account we pick has a `drawerOnSelection`
+                where: { $0.accountPickerAccount.drawerOnSelection != nil }
+            ),
+            let drawerOnSelection = selectedAccount.accountPickerAccount.drawerOnSelection,
+            IsAccountUpdateRequired(forAccount: selectedAccount.partnerAccount)
+        {
+            footerView?.showLoadingView(true)
+            dataSource
+                .markConsentAcquired()
+                .observe { [weak self] _ in
+                    guard let self = self else { return }
+                    footerView?.showLoadingView(false)
+                    self.presentAccountUpdateRequiredDrawer(
+                        drawerOnSelection: drawerOnSelection,
+                        partnerAccount: selectedAccount.partnerAccount
+                    )
```

Comment:
the logic here is a "reversal" of what happened here:

this code is a follow up from here:

https://github.com/stripe/stripe-ios/pull/3870/files?diff=split&w=1#r1701045857